jupyter_client/jsonutil.py

```diff
@@ -106,7 +109,17 @@ def json_default(obj):
     if isinstance(obj, datetime):
         obj = _ensure_tzinfo(obj)
         return obj.isoformat().replace('+00:00', 'Z')
-    elif isinstance(obj, bytes):
+
+    if isinstance(obj, bytes):
         return b2a_base64(obj).decode('ascii')
-    else:
-        raise TypeError("%r is not JSON serializable" % obj)
+
+    if isinstance(obj, (set, types.GeneratorType, Iterable)):
```

Comment:
Sets and generators are iterables.
```suggestion
    if isinstance(obj, Iterable):
```

Reply:
Thanks!